### ISSUE TITLE
Work Profile Setup works in Insular, but not in Shelter

### ISSUE THREAD

**Issue by Rafee-M** [PeterCxy/Shelter#302]
When trying to setup work profile, Shelter is unable to do so, but Insular can do it successfully

Steps tried:

1. Set Shelter as device admin

Questions:

Info: I have debloated my phone a little (not all, just basic telemetry packages via https://github.com/Universal-Debloater-Alliance/universal-android-debloater-next-generation) and I am currently using microG instead of Google Play Services. Oxygen OS 13, Android 13.

1. Although I doubt the packages I disabled would have an effect on the work profile, but I tried with disabling them as well.
2. Does Shelter depend on Google Play Services somehow?
3. How does the setup process of Shelter differ from Insular that Insular and Shelter doesn't?

**Comment by Rafee-M** [comment]
Update:

Shelter is somehow dependent on Google Play Services


Steps to reproduce:

1. Another phone (similarly debloated except for Google Play Services not swapped for microG) with OxygenOS 13, Android 13 as well
2. Insular & Shelter both work
3. Disable Play Services
4. Only Insular works

Video:


https://github.com/PeterCxy/Shelter/assets/69535896/68ae9491-dc6f-401d-9b3b-cc7559d06b99

**Comment by Xanatos00** [comment]
Indeed, throughout my experiments, I found that Shelter needs following packages, for the initial configuration:
- com.google.android.gms (|Google Play services)
- com.google.android.gsf(Google Services Framework)

**Comment by gbraad** [comment]
I disable "Play Store" in the work profile and immediately Shelter lost reference, and on start wanted to create a new Work profile. So yes, it depends on Google Play services... even after re-enabling this in the profile, Shelter does not recognize the profile.
